backend/routes.js

```diff
@@ -447,6 +447,99 @@ route.POST('/deleteFile/{hash}', {
   return h.response()
 })
 
+route.POST('/kv/{contractID}/{key}', {
+  auth: {
+    strategies: ['chel-shelter'],
+    mode: 'required'
+  },
+  payload: {
+    parse: false,
+    maxBytes: 6 * MEGABYTE, // TODO: make this a configurable setting
+    timeout: 10 * SECOND // TODO: make this a configurable setting
+  }
+}, async function (request, h) {
+  const { contractID, key } = request.params
+  console.debug(`POST /kv/${contractID}/${key}`)
+
+  if (key.startsWith('_private')) {
+    return Boom.notFound()
+  }
+
+  if (!ctEq(request.auth.credentials.billableContractID, contractID)) {
+    return Boom.unauthorized(null, 'shelter')
+  }
+
+  const existing = await sbp('chelonia/db/get', `_private_kv_${contractID}_${key}`)
+
+  // Some protection against accidental overwriting by implementing the if-match
+  // header
+  if (request.headers['if-match']) {
+    if (!existing) {
+      return Boom.preconditionFailed()
+    }
+    const expectedEtag = request.headers['if-match']
+    if (expectedEtag === '*') {
+      // pass through
+    } else {
+      // "Quote" string
+      const cid = JSON.stringify(createCID(existing))
+      if (!expectedEtag.split(',').map(v => v.trim()).includes(cid)) {
+        return Boom.preconditionFailed()
+      }
+    }
+  }
+
+  try {
+    const serializedData = JSON.parse(request.payload.toString())
+    const { contracts } = sbp('chelonia/rootState')
+    // Check that the height is the latest value
```

Comment:
Could you add a comment here explaining why this height check is being done? It's not completely clear, but I'm guessing it's to make sure that we're using the latest keys for the signature